tracee-ebpf/tracee/argprinters.go

```diff
@@ -54,6 +54,48 @@ func (t *Tracee) prepareArgs(ctx *context, args map[string]interface{}) error {
 			args[key] = fmt.Sprintf("0x%X", ptr)
 		}
 	}
+
+	switch ctx.EventID {
+	case ConnectEventID, AcceptEventID, Accept4EventID, BindEventID, GetsocknameEventID:
+		if sockAddr, isStrMap := args["addr"].(map[string]string); isStrMap {
+			var s string
+			for key, val := range sockAddr {
+				s += fmt.Sprintf("'%s': '%s',", key, val)
+			}
+			s = strings.TrimSuffix(s, ",")
+			s = fmt.Sprintf("{%s}", s)
+			args["addr"] = s
+		}
+	case SecuritySocketBindEventID, SecuritySocketAcceptEventID, SecuritySocketListenEventID:
+		if sockAddr, isStrMap := args["local_addr"].(map[string]string); isStrMap {
+			var s string
+			for key, val := range sockAddr {
+				s += fmt.Sprintf("'%s': '%s',", key, val)
+			}
+			s = strings.TrimSuffix(s, ",")
+			s = fmt.Sprintf("{%s}", s)
+			args["local_addr"] = s
+		}
+	case SecuritySocketConnectEventID, SocketDupEventID:
+		if sockAddr, isStrMap := args["remote_addr"].(map[string]string); isStrMap {
+			var s string
+			for key, val := range sockAddr {
+				s += fmt.Sprintf("'%s': '%s',", key, val)
+			}
+			s = strings.TrimSuffix(s, ",")
+			s = fmt.Sprintf("{%s}", s)
+			args["remote_addr"] = s
+		}
+	case MemProtAlertEventID:
+		if alert, isAlert := args["alert"].(alert); isAlert {
+			args["alert"] = PrintAlert(alert)
+		}
+	}
```

Comment:
As a future PR, we should think about taking this code to be part of the socket/alert decoding